datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -2130,16 +2130,10 @@ impl Filter {
             }
         }
 
-        // filter predicates should not be aliased
-        if let Expr::Alias(Alias { expr, name, .. }) = predicate {
-            return plan_err!(
-                "Attempted to create Filter predicate with \
-                expression `{expr}` aliased as '{name}'. Filter predicates should not be \
-                aliased."
-            );
-        }
-
-        Ok(Self { predicate, input })
+        Ok(Self {
+            predicate: predicate.unalias(),
```

Comment:
After unalias, it might still be another alias, which doesn't seem very thorough.
Therefore, what do you think about executing `unalias_nested` during planning, like [this](https://github.com/jonahgao/datafusion/commit/7c924284553920004059717c1bac68d913cd6daf)?
This should be able to solve the current problem.

 As for whether we need to strictly strip aliases, perhaps we can discuss that separately.

Reply:
done.